<?xml version="1.0"?>
<!--***********************************************************
	MoveIt Filter Joint States Node
	Emily Sheetz, NSTGRO VTE 2023

	run:
		$ roslaunch val_moveit_planner_executor moveit_filter_joint_states_node.launch
************************************************************-->
<launch>

	<!-- parameters for specifying joint state topics -->
	<arg name="filter_js_topic" default="/moveit_filtered_joint_states"/>
	<arg name="js_topic" default="/ihmc_ros/valkyrie/output/joint_states"/>

	<!-- parameters for specifying joints to filter out -->
	<arg name="filter_joints" default="[hokuyo_joint,
		rightThumbRoll,
	  	rightThumbPitch1,
	  	rightThumbPitch2,
	  	rightThumbPitch3,
	  	rightIndexFingerPitch1,
	  	rightIndexFingerPitch2,
	  	rightIndexFingerPitch3,
	  	rightMiddleFingerPitch1,
	  	rightMiddleFingerPitch2,
	  	rightMiddleFingerPitch3,
	  	rightPinkyPitch1,
	  	rightPinkyPitch2,
	  	rightPinkyPitch3,
	  	leftThumbRoll,
	  	leftThumbPitch1,
	  	leftThumbPitch2,
	  	leftThumbPitch3,
	  	leftIndexFingerPitch1,
	  	leftIndexFingerPitch2,
	  	leftIndexFingerPitch3,
	  	leftMiddleFingerPitch1,
	  	leftMiddleFingerPitch2,
	  	leftMiddleFingerPitch3,
	  	leftPinkyPitch1,
	  	leftPinkyPitch2,
	  	leftPinkyPitch3]"
	/>


	<rosparam param="filter_joints" subst_value="True">$(arg filter_joints)</rosparam>

	<node pkg="val_moveit_planner_executor" type="moveit_filter_joint_states_node" name="ValkyrieMoveItFilterJointStatesNode" output="screen">
		<param name="filter_js_topic" type="string" value="$(arg filter_js_topic)"/>
		<param name="js_topic" type="string" value="$(arg js_topic)"/>
		<!--<param name="" type="" value=""/> -->
	</node>

</launch>
